packages/react-badge/src/components/Badge/useBadgeStyles.ts

```diff
@@ -276,7 +294,16 @@ export const useBadgeStyles = (state: BadgeState): BadgeState => {
   );
 
   if (state.icon) {
-    state.icon.className = mergeClasses(styles.icon, state.icon.className);
+    state.icon.className = mergeClasses(
+      styles.icon,
+      state.size === 'tiny' && styles.iconTiny,
+      state.size === 'extra-small' && styles.iconExtraSmall,
+      state.size === 'small' && styles.iconSmall,
+      state.size === 'medium' && styles.iconMedium,
+      state.size === 'large' && styles.iconLarge,
+      state.size === 'extra-large' && styles.iconExtraLarge,
```

Comment:
Should we spin these styles in a separate `useIconStyles` call so that we could do `iconStyles[state.size]` here instead of all these checks?

Reply:
Sure. In fact, this whole styles file could be simplified by indexing on the various state props instead of having a bunch of individual checks. Since that's a bit of a bigger change and out of scope of this PR, I'll do it in a follow-up PR.

Reply:
#21271